client/src/pages/Actions.ts

```diff
@@ -134,6 +152,98 @@ export const getOrders = async () => {
   return (await fetchJson("GET", {}, ORDER_LIST_API, true)) || [];
 };
 
+/**
+ * Retrieves the list of items in an order.
+ *
+ * @param {string} orderName - The Spot order name in the form of {merchants/*\/orders/*}.
+ * @returns {CartItem[]} contents - The items in the order.
+ */
+export const getOrderContents = async (orderName: string) => {
+  // parse merchant ID from order name
+  let merchantId = "";
+  const matches = orderName.match(/(?<=merchants\/)(.*?)(?=\/orders)/gi);
+  if (matches) {
+    merchantId = matches[0];
+  }
+
+  // retrieve order items
+  const order = await fetchJson("GET", {}, `${ORDER_API}/${orderName}`, true);
+  let contents: CartItem[] = [];
+  if (order?.items) {
+    const itemBarcodes = order.items.map(
+      (orderItem: OrderItem) => orderItem.subtitle
+    );
+    const data = {
+      "merchant-id": merchantId,
+      barcode: itemBarcodes,
+    };
+    const items = await fetchJson("POST", data, ITEM_LIST_API);
+    contents = order.items.map((orderItem: OrderItem) => {
+      return {
+        item: items.filter((item: Item) => item.barcode == orderItem.subtitle),
+        quantity: orderItem.quantity,
+      };
+    });
+  }
+  return contents;
+};
+
+/**
+ * Creates an order in the Spot database.
+ * Returns the Spot Order response if successful.
+ *
+ * @param {Store} store - The store at which the order is made.
+ * @param {CartItem[]} cartItems - The list of items in the order.
+ */
+export const createOrder = async (store: Store, cartItems: CartItem[]) => {
+  if (isWeb) {
+    return { name: TEST_ORDER_NAME, id: TEST_ORDER_ID }; // placeholder order for web flow
+  }
+  // TODO (#): add currency code in items and price utility functions
+  const currencyCode = DEFAULT_CURRENCY_CODE;
+  const orderReq = {
+    title: `Order @ ${store.name}`,
+    items: cartItems.map((cartItem) => {
+      return {
+        title: cartItem.item.name,
+        subtitle: cartItem.item.barcode,
+        quantity: cartItem.quantity,
+        price: {
+          currency: currencyCode,
+          value: cartItem.item.price.toFixed(PRICE_FRACTION_DIGITS),
+        },
+      };
+    }),
+    total: {
+      currency: currencyCode,
+      value: getTotalPrice(cartItems),
+    },
+    status: { type: "COMPLETED" },
+  };
+
+  let orderRes = await microapps
+    .createOrder(orderReq)
+    .catch((err: any) => console.error(err));
+
+  if (orderRes) {
+    // update the order actions for the newly created order
+    orderRes = await fetchJson(
+      "POST",
+      { orderName: orderRes.name },
+      ORDER_UPDATE_API,
+      true
+    );
+
+    // add the order to the ScanAndGo database
+    const data = {
+      merchantId: process.env.REACT_APP_SPOT_MERCHANT_ID,
+      orderId: orderRes.orderId,
+    };
+    await fetchText("POST", data, ORDER_ADD_API, true);
```

Comment:
How come ORDER_UPDATE_API - called before ORDER_CREATE?

Reply:
`ORDER_UPDATE_API` updates the Spot Order, which we created in `microapps.createOrder()`, so that users can go to the receipt page/view the order details from the chathead. `ORDER_ADD_API` makes a record of the order in our own databases, which is just for logging/debugging purposes for now but we might use in the future e.g. to allow merchants to download their ScanAndGo orders.

Reply:
Got it!

Reply:
Perhapps name the API better like MICROAPPS_ORDER_UPDATE? to make it clear

Reply:
good idea!